packages/controller/src/Controller.ts

```diff
@@ -766,13 +841,15 @@ export default class Controller {
 				this.instanceUpdated(instance);
 			}
 
+			this.instancesDirty = true;
 			lib.invokeHook(this.plugins, "onInstanceConfigFieldChanged", instance, group, field, prev);
 		});
 
 		this.instanceUpdated(instance);
 	}
 
 	instanceUpdated(instance: InstanceInfo) {
+		this.instancesDirty = true;
```

Comment:
`instanceUpdated` is called by `addInstanceHooks` which is called by `_startInternal` right after the config is loaded. Therefore, after every startup the instances will be considered dirty, is this the expected behaviour? Neither hosts nor modpacks have this behaviour.

Reply:
No this is not expected behaviour. I looked into this and found that instanceUpdated should really be instanceDetailsUpdated, and did some refactoring to make instances not be marked as dirty on startup.